crates/network/src/lib.rs

```diff
@@ -0,0 +1,386 @@
+use std::{fmt::Debug, net::SocketAddr, sync::Arc, time::Duration};
+
+use bevy_app::{App, Plugin, PostStartup, PreUpdate, Update};
+use bevy_ecs::{
+    event::EventWriter,
+    prelude::{Commands, Component, Entity, Event, EventReader, Query, Res, World},
+    system::{Resource, SystemState},
+};
+use bytes::{Bytes, BytesMut};
+use flume::{Receiver, Sender, TryRecvError, TrySendError};
+use rjacraft_protocol::{
+    packets::client::{ClientHandshakePacket, ClientStatusPacket, HandshakeState},
+    Decoder, Encoder,
+};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    net::{TcpListener, TcpStream, ToSocketAddrs},
+    runtime::Runtime,
+    task::JoinHandle,
+    time::Instant,
+};
+
+use crate::{decode::PacketDecoder, encode::PacketEncoder};
```

Comment:
```suggestion
use self::{decode::PacketDecoder, encode::PacketEncoder};
```

This would be better since the imports would be relative to current mod thus making the code more explicit.